src/qtt/algorithms/fitting.py

```diff
@@ -171,20 +172,20 @@ def fitFermiLinear(x_data, y_data, verbose=0, fig=None, lever_arm=1.16, l=None,
 
     .. seealso:: FermiLinear
     """
-    xdata = np.array(x_data)
-    ydata = np.array(y_data)
+    x_data = np.array(x_data)
+    y_data = np.array(y_data)
 
     if l is not None:
         warnings.warn('use argument lever_arm instead of l')
         lever_arm = l
 
     # initial values
-    linear_part, fermi_part = initFermiLinear(xdata, ydata, fig=None)
+    linear_part, fermi_part = initFermiLinear(x_data, y_data, fig=None)
     initial_parameters = linear_part + fermi_part
 
     # fit
-    def fermi_linear_fitting_function(xdata, a, b, cc, A, T):
-        return FermiLinear(xdata, a, b, cc, A, T, l=lever_arm)
+    def fermi_linear_fitting_function(x_data, a, b, cc, A, T):
```

Comment:
The same comment is still valid. You are shadowing a variable in outer scope. x_data is already used in the outer method. You have to look twice to know to which x_data in line 188 refers to.
Because the only thing is that FermiLinear is called with an extra argument. Maybe you could use partial instead  partial(FermiLinear, l=lever_arm) instead of fermi_linear_fitting_function?

Reply:
Ah I see the point now. I will use `partial`

Reply:
The `partial` does not work, because `partial(FermiLinear, l=lever_arm)` does not have `__name__` which is required by lmfit.

I am therefore still using the old imlementation, which seems the cleanest solution.